subxt/src/backend/chain_head/follow_stream.rs

```diff
@@ -154,9 +160,16 @@ impl<Hash> Stream for FollowStream<Hash> {
                                 continue;
                             }
 
-                            // Finish forever if there's an error, passing it on.
-                            this.stream = InnerStreamState::Finished;
-                            return Poll::Ready(Some(Err(e)));
+                            // Finish forever if there's an error which is done as follows:
+                            //
+                            // 1) Send the FollowEvent::BackendClosed message which tells to subscriber that subscription is closed.
+                            // 2) Send the error on the backend driver stream.
+                            // 3) Finish the stream.
+                            //
+                            this.stream = InnerStreamState::Error(Some(e));
+                            return Poll::Ready(Some(Ok(FollowStreamMsg::Event(
+                                FollowEvent::BackendClosed,
+                            ))));
```

Comment:
Ah, this is where we turn any error that's not a `DisconnectedWillReconnect` into something which closes the subscription, so RpcClients don't need to know anything special!